homeassistant/components/plex/server.py

```diff
@@ -602,11 +603,46 @@ def lookup_tv():
                 )
                 return None
 
+        def lookup_movie():
```

Comment:
Let's not define this inline.

Reply:
Moved all previously inline lookup methods to a separate file. Only modification to those methods was to use a new custom exception to pass the failed lookup details back to a central logging call.